Source/RapyutaSimulationPlugins/Public/Core/RRThreadUtils.h

```diff
@@ -82,4 +93,51 @@ class RAPYUTASIMULATIONPLUGINS_API URRThreadUtils : public UBlueprintFunctionLib
             MoveTemp(InTask),
             TUniqueFunction<void()>([InCompletionCallback = MoveTemp(InCompletionCallback)]() { InCompletionCallback(); }));
     }
+    template<typename TResult>
+    static auto AddAsyncTaskInThreadPool(FRRAsyncJob& OutAsyncJob,

```

Comment:
This function returns nothing, so the return type will always default to void. I think it's better to be explicit and change it.